pkg/cli/create/routeedge.go

```diff
@@ -104,6 +104,12 @@ func (o *CreateEdgeRouteOptions) Run() error {
 		return err
 	}
 
+	// If the namespace is not the default, add it. This makes sure it will eventually
+	// appear in the manifest if the dry-run is enabled.
+	if o.CreateRouteSubcommandOptions.EnforceNamespace {
```

Comment:
Better, but still I mentioned to move this if-clause to `UnsecuredRoute` method invoked above. This way it'll apply to all invocations.